1. Installed the dependencies to run the service
2. cd api
3. Got issue while Starting the service itself which are mentioned below with solution i did:
    A. running "node adonis migration:run"
        Error: ER_NOT_SUPPORTED_AUTH_MODE: Client does not support authentication protocol requested by server; consider upgrading MySQL client

        Solution
        on Terminal
        sudo mysql
        ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY 'HoldOn#6!';
        FLUSH PRIVILEGES;
        quit

    B. After 3 got this issue
        Error: ER_BAD_DB_ERROR: Unknown database 'adonis'

        Solution
        mysql -u root -p
        HoldOn#6!
        CREATE DATABASE adonis;
        quit

    C. The read me file does not contains command to start the node service

        Solution
        npm start

4. Automating "/api/v1/register"

    NO issue found in api with relates to API doc
    You will see this test case in automation

5. Automating "/api/v1/login"

   NO issue found in api with relates to API doc
   You will see this test case in automation

6. Automating "/api/v1/users/create"

    Issue 1 : In doc it is mentioned that the route is "/api/v1/users/create" when we use this we are getting 404 not found that means this endpoint does not exist
    Solution: Check Route.js and there we can see correct route is "employees/create"

    Issue 2 : After solving issue 1 we are getting 401 unauthorized while hitting request
    Solution: In the doc it is not mentioned that this route needs a authorization. Used the token with bearer which we got after login and was able to fix this issue
    You will see this test case in automation

    Issue 3: Getting 400 Bad request error after using fresh bearer token
        {
        "status": "error",
        "data": {
            "code": "ER_NO_DEFAULT_FOR_FIELD",
            "errno": 1364,
            "sqlMessage": "Field 'employee_name' doesn't have a default value",
            "sqlState": "HY000",
            "index": 0,
            "sql": "insert into `employees` (`company_id`, `created_at`, `employee_email`, `id`, `updated_at`, `user_id`) values (21, '2020-10-11 14:24:02', 'sarvesh.singh811987226787656222@thinxnet.com', 225111, '2020-10-11 14:24:02', 36)"
        },
        "message": "Somethign went wrong! Please try again later"
    }

    Solution : You have to follow the below shared rules while creating employee and sharing a sample request below
    1. While creating employee the employee_email has to be unique so you do not get the Duplicate user_email issue
    2. You have to pass employee_name field while creating employee to avoid default value not found error for employee_name
        {
            "profile": {
                "role": "admin",
                "id": 10,
                "employee_email": "sarvesh.singh6@thinxnet.com",
                "employee_name": "Sarvesh",
                "password": "Yellow*99"
            }
        }
        Now we are able to create employee with this correct request body, So i will show this error in automation first and then will create employee with correct request body
        You will see this test case in automation

8. Automating Edit profile

    We need to login with the employee credentials and use that toke to get the profile
    You will see this test case in automation

    Issue 1: There is no route available for edit profile in route.js hence we cannot update employee profile and thus api is giving 204.
    You will see this test case in automation



